### ISSUE TITLE
body.as[Json] breaks on null values

### ISSUE THREAD

**Issue by lderks** [finagle/finch#222]
I have extended the demo project to accept some json that is being posted. See the code below. This works fine.

Except when I post json with a null value (which is still valid json), the server returns a 500 Internal Server Error. Also this error is not handled by the case _ => InternalServerError() in the handleExceptions method. 

json:
```json
{
    "name": null
}
```

RequestReader[Json]:
```scala
object JsonPost extends Service[AuthRequest, Json] {
    def apply(req: AuthRequest): Future[Json] = for {
      in <- json(req)
    } yield in
  }
  val json:RequestReader[Json] = body.as[Json]
```

**Comment by lderks** [comment]
Actually I think it is the TurnIntoHttp[Json] which can't handle the Json with null values

**Comment by vkostyukov** [comment]
Thanks @lderks for the report! I probably found a cause for this.

**Comment by vkostyukov** [comment]
Fix is deployed as `0.7.0-SNAPSHOT`.
